resolwe_bio/processes/expression/featureCounts.py

```diff
@@ -0,0 +1,888 @@
+
+        if inputs.read_filtering.primary:
+            args.append("--primary")
+
+        if inputs.read_filtering.ignore_dup:
+            args.append("--ignoreDup")
+
+        if inputs.exon_exon_junctions.junc_counts:
+            args.append("-J")
+
+        if inputs.exon_exon_junctions.junc_counts and inputs.exon_exon_junctions.genome:
+            args.extend(["-G", inputs.exon_exon_junctions.genome.path])
+
+        if inputs.general.by_read_group:
+            args.append("--byReadGroup")
+
+        if inputs.general.count_long_reads:
+            args.append("-L")
+
+        if inputs.miscellaneous.report_reads:
+            args.append("-R CORE")
+
+        if inputs.miscellaneous.verbose:
+            args.append("--verbose")
+
+        # for paired-end samples, extend the args with paired-end input options
+        if paired_end and inputs.paired_end.is_paired_end:
+            args.append("-p")
+
+        if paired_end and inputs.paired_end.require_both_ends_mapped:
+            args.append("-B")
+
+        if paired_end and inputs.paired_end.check_frag_length:
+            args.append("-P")
+
+        if paired_end and inputs.paired_end.check_frag_length:
+            args.extend(["-d", inputs.paired_end.min_frag_length])
+
+        if paired_end and inputs.paired_end.check_frag_length:
+            args.extend(["-D", inputs.paired_end.max_frag_length])
+
+        if paired_end and inputs.paired_end.do_not_count_chimeric_fragments:
+            args.append("-C")
+
+        if paired_end and inputs.paired_end.do_not_sort:
+            args.append("--do_not_sort")
+
+        # Run featureCounts
+        return_code, _, _ = Cmd["featureCounts"][args][bam_file] & TEE(retcode=None)
+        if return_code:
+            self.error("Error while running featureCounts.")
+
+        # parse featureCounts output
+        get_gene_counts("featureCounts_rc.txt", "rc.txt", bam_file)
+        get_gene_lenghts("featureCounts_rc.txt", "gene_lengths.txt")
+
+        # Normalize counts
+        rnanorm_args = [
+            "rc.txt",
+            "--gene-lengths",
+            "gene_lengths.txt",
+            "--tpm-output",
+            "tpm.txt",
+            "--cpm-output",
+            "cpm.txt",
+        ]
+        return_code, _, _ = Cmd["rnanorm"][rnanorm_args] & TEE(retcode=None)
+        if return_code:
+            self.error("Error while normalizing counts using rnanorm.")
+
+        # prepare the expression set outputs
+        feature_ids = pd.read_csv(
+            "rc.txt", sep="\t", index_col="FEATURE_ID"
+        ).index.to_list()
+
+        feature_filters = {
+            "source": inputs.annotation.output.source,
+            "species": inputs.aligned_reads.output.species,
+            "feature_id__in": feature_ids,
+        }
+
+        feature_ids_to_names = {
+            f.feature_id: f.name for f in self.feature.filter(**feature_filters)
+        }
+
+        prepare_expression_set(
+            rc="rc.txt",
```

Comment:
See how nicely readable explicitly named arguments are? :)

Reply:
> Can you explain how the test files changed? I would expect that results to be identical to the yaml process version.

In the YAML process, the float values were rounded at ~10th decimal place when importing tab delimited data to Pandas when creating "expression set" files. As explained in my comment to @JureZmrzlikar, we now use  `float_precission="round_trip"` so that the values don't change when reading and writing exiting float values using Pandas.